app/questionnaire/navigation.py

```diff
@@ -179,3 +191,11 @@ def _generate_link_names(self, repeating_rule):
             link_names[link_name] = " ".join(link_names[link_name])
 
         return link_names
+
+    @staticmethod
+    def _all_sections_completed(navigation):
```

Comment:
We shouldn't rely on nav_items being completed to indicate all sections are complete incase there are groups that are hidden and not complete. We should check all blocks in the routing path are in completed blocks